src/realm/object-store/sync/sync_session.cpp

```diff
@@ -468,15 +468,37 @@ void SyncSession::download_fresh_realm(sync::ProtocolErrorInfo::Action server_re
     }
 
     fresh_sync_session->assert_mutex_unlocked();
-    if (auto local_subs_store = m_flx_subscription_store) {
-        sync::SubscriptionSet active = local_subs_store->get_active();
-        auto fresh_sub_store = fresh_sync_session->get_flx_subscription_store();
-        REALM_ASSERT(fresh_sub_store);
-        auto fresh_mut_sub = fresh_sub_store->get_latest().make_mutable_copy();
-        fresh_mut_sub.import(active);
-        std::move(fresh_mut_sub)
-            .commit()
-            .get_state_change_notification(sync::SubscriptionSet::State::Complete)
+    // The fresh realm uses flexible sync.
+    if (auto fresh_sub_store = fresh_sync_session->get_flx_subscription_store()) {
+        auto fresh_sub = fresh_sub_store->get_latest();
+        // The local realm uses flexible sync as well so copy the active subscription set to the fresh realm.
+        if (auto local_subs_store = m_flx_subscription_store) {
+            sync::SubscriptionSet active = local_subs_store->get_active();
+            auto fresh_mut_sub = fresh_sub.make_mutable_copy();
+            fresh_mut_sub.import(active);
+            fresh_sub = fresh_mut_sub.commit();
+        }
+        fresh_sub.get_state_change_notification(sync::SubscriptionSet::State::Complete)
+            .then([=](sync::SubscriptionSet::State state) {
+                // Create subscriptions in the fresh realm based on the schema instructions received in the bootstrap
+                // message.
+                if (server_requests_action != sync::ProtocolErrorInfo::Action::MigrateToFLX ||
+                    !m_migration_store->is_migration_in_progress()) {
+                    return util::Future<sync::SubscriptionSet::State>::make_ready(state);
+                }
+
+                fresh_sync_session->m_migration_store->create_subscriptions(*fresh_sub_store,
+                                                                            m_migration_store->get_query_string());
+                auto latest_subs = fresh_sub_store->get_latest();
+                {
+                    util::CheckedLockGuard lock(m_state_mutex);
+                    // Save a copy of the subscriptions so we add them to the local realm once the
+                    // subscription store is created.
+                    m_active_subscriptions_after_migration = latest_subs;
+                }
```

Comment:
Will this perform a change of query for the generated subscriptions and a subsequent bootstrap on the "fresh" SyncSession before the client reset is complete? And then the saved `m_active_subscriptions_after_migration` will be applied/migrated to the subscription store in the original SyncSession by `make_active_subscription_set()` when its subscription store is created? (Just clarifying my understanding of the events that will take place.)

Reply:
yes, that's the exact chain of events.